numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -2064,6 +2094,91 @@ linear_search_type_resolver(PyUFuncObject *self,
     return -1;
 }
 
+
+static int
+type_tuple_type_resolver_core(PyUFuncObject *self,
+        PyArrayObject **op,
+        NPY_CASTING input_casting, NPY_CASTING casting,
+        int specified_types[],
+        int any_object,
+        int no_castable_output, int use_min_scalar,
+        PyArray_Descr **out_dtype)
+{
+    int i, j;
+    int nop = self->nargs;
+    int types[NPY_MAXARGS];
+
+    /* For making a better error message on coercion error */
+    char err_dst_typecode = '-', err_src_typecode = '-';
+
+    /* If the ufunc has userloops, search for them. */
+    if (self->userloops) {
+        switch (type_tuple_userloop_type_resolver(self,
+                nop, specified_types,
+                op, input_casting, casting,
+                any_object, use_min_scalar,
+                out_dtype)) {
+            /* Error */
+            case -1:
+                return -1;
+                /* Found matching loop */
+            case 1:
+                return 0;
+        }
+    }
+
+    for (i = 0; i < self->ntypes; ++i) {
+        char *orig_types = self->types + i*self->nargs;
+
+        /* Check specified types and copy into an int array for matching */
```

Comment:
Maybe note this code appears twice and could be put into a small function?

Reply:
Will add a note that its duplicated.  Its not quite trivial to refactor because `orig_types` is `char *` in one case and `int *`.